app/admin/vacation.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+ActiveAdmin.register Vacation do
+  permit_params :start_date, :end_date, :status, :user_id, :commercial_approver_id, :administrative_approver_id
+
+  menu parent: User.model_name.human(count: 2)
+  
+  filter :start_date
+  filter :end_date
+  filter :user, as: :select, collection: -> { User.includes(:vacations).engineer.active.where.not(vacations: { user: nil } ) }
+
+  member_action :approve, method: :put do
+    resource.set_approver(current_user)
+    resource.update(status: :approved) unless resource.commercial_approver.nil? && resource.administrative_approver.nil?
```

Comment:
we should add permissions here to verify this

Reply:
the permissions in the ability file are different from the ones we need here, we need permissions here based in the admin role (we need to create those roles)